services/UserService_test.go

```diff
@@ -0,0 +1,705 @@
+					})
+
+					ginkgo.It("should merge data of given user", func() {
+						gomega.Expect(err).Should(gomega.BeNil()) // no error during BeforeEach
+
+						err = client.UserService.Update(context.TODO(), *user, *userBlueprint, services.Merge)
+						gomega.Expect(err).ShouldNot(gomega.BeNil())
+					})
+				})
+
+				ginkgo.When("when replace data of given user", func() {
+					ginkgo.BeforeEach(func() {
+						recName = userUpdateEmptyReplace
+					})
+
+					ginkgo.It("should replace data of given user", func() {
+						gomega.Expect(err).Should(gomega.BeNil()) // no error during BeforeEach
+
+						err = client.UserService.Update(context.TODO(), *user, *userBlueprint, services.Replace)
+						gomega.Expect(err).ShouldNot(gomega.BeNil())
+					})
+				})
+			})
+		})
+
+		ginkgo.Context("when user doesn't exist", func() {
+			ginkgo.BeforeEach(func() {
+				recName = userUpdateUnknownUser
+
+				user = resources.CreateUser(idNonExistingUser)
+				if user == nil {
+					err = fmt.Errorf("no user to finish test")
+				}
+
+				userBlueprint = blueprint.CreateUserBlueprint()
+				if userBlueprint == nil {
+					err = fmt.Errorf("no user blueprint to finish test")
+					return
+				}
+				userBlueprint.SetEmail("pancake@pizza.com")
+			})
+
+			ginkgo.It("should return that user with given ID doesn't exist", func() {
+				gomega.Expect(err).Should(gomega.BeNil()) // no error during BeforeEach
+
+				err = client.UserService.Update(context.TODO(), *user, *userBlueprint, services.Merge)
+				gomega.Expect(err).ShouldNot(gomega.BeNil())
+			})
+		})
+
+		ginkgo.Context("when user is empty", func() {
+			ginkgo.BeforeEach(func() {
+				recName = userUpdateNoUser
+
+				userBlueprint = blueprint.CreateUserBlueprint()
+				if userBlueprint == nil {
+					err = fmt.Errorf("no user blueprint to finish test")
+					return
+				}
+				userBlueprint.SetEmail("pancake@pizza.com")
+			})
+
+			ginkgo.It("should return that user has no ID", func() {
+				gomega.Expect(err).Should(gomega.BeNil()) // no error during BeforeEach
+
+				err = client.UserService.Update(context.TODO(), resources.User{}, *userBlueprint, services.Merge)
+				gomega.Expect(err).ShouldNot(gomega.BeNil())
+			})
+		})
+	})
+
+	ginkgo.Describe("change user authentication driver", func() {
+		var user *resources.User
+
+		ginkgo.Context("when user exists", func() {
+			ginkgo.BeforeEach(func() {
+				user = resources.CreateUser(idExistingUser)
+				if user == nil {
+					err = fmt.Errorf("no user to finish test")
+				}
+			})
+
+			ginkgo.When("when authentication driver is not empty", func() {
+				ginkgo.BeforeEach(func() {
+					recName = userAuthDriver
+				})
+
+				ginkgo.It("should change authentication driver of given user", func() {
+					gomega.Expect(err).Should(gomega.BeNil()) // no error during BeforeEach
+
+					err = client.UserService.ChangeAuthDriver(context.TODO(), *user, existingAuthDriver)
+					gomega.Expect(err).Should(gomega.BeNil())
```

Comment:
You have to make sure the data was updated as you expected. Ok, just do it everywhere where you change something :smile: